python/pylibcugraph/pylibcugraph/node2vec.pyx

```diff
@@ -0,0 +1,183 @@
+        and running algorithms.
+
+    graph : SGGraph
+        The input graph.
+
+    seed_array: device array type
+        Device array containing the pointer to the array of seed vertices.
+
+    max_depth : size_t
+        Maximum number of vertices in generated path
+
+    compress_result : bool_t
+        If true, the third return device array contains the sizes for each path,
+        otherwise outputs empty device array.
+
+    p : double
+        The return factor p represents the likelihood of backtracking to a node
+        in the walk. A higher value (> max(q, 1)) makes it less likely to sample
+        a previously visited node, while a lower value (< min(q, 1)) would make it
+        more likely to backtrack, making the walk more "local".
+
+    q : double
+        The in-out factor q represents the likelihood of visiting nodes closer or
+        further from the outgoing node. If q > 1, the random walk is likelier to
+        visit nodes closer to the outgoing node. If q < 1, the random walk is
+        likelier to visit nodes further from the outgoing node.
+
+    Returns
+    -------
+    A tuple of device arrays, where the first item in the tuple is a device
+    array containing the compressed paths, the second item is a device
+    array containing the corresponding weights for each edge traversed in
+    each path, and the third item is a device array containing the sizes
+    for each of the compressed paths, if compress_result is True.
+
+    Examples
+    --------
+    >>> import pylibcugraph, cupy, numpy
+    >>> srcs = cupy.asarray([0, 1, 2], dtype=numpy.int32)
+    >>> dsts = cupy.asarray([1, 2, 3], dtype=numpy.int32)
+    >>> seeds = cupy.asarrray([0, 0, 1], dtype=numpy.int32)
+    >>> weights = cupy.asarray([1.0, 1.0, 1.0], dtype=numpy.float32)
+    >>> resource_handle = pylibcugraph.experimental.ResourceHandle()
+    >>> graph_props = pylibcugraph.experimental.GraphProperties(
+    ...     is_symmetric=False, is_multigraph=False)
+    >>> G = pylibcugraph.experimental.SGGraph(
+    ...     resource_handle, graph_props, srcs, dsts, weights,
+    ...     store_transposed=False, renumber=False, do_expensive_check=False)
+    >>> (paths, weights, sizes) = pylibcugraph.experimental.node2vec(
+    ...                             resource_handle, G, seeds, 3, True, 1.0, 1.0)
+
+    """
+
+    # FIXME: import these modules here for now until a better pattern can be
+    # used for optional imports (perhaps 'import_optional()' from cugraph), or
+    # these are made hard dependencies.
+    try:
+        import cupy
+    except ModuleNotFoundError:
+        raise RuntimeError("node2vec requires the cupy package, which could not "
+                           "be imported")
+    try:
+        import numpy
+    except ModuleNotFoundError:
+        raise RuntimeError("node2vec requires the numpy package, which could not "
+                           "be imported")
```

Comment:
It doesn't look like you're using `numpy` in this function. If that's the case, then this should be removed.

Reply:
I was looking at the doctest and thought "I need to import numpy still". I didn't, and it's been changed.